Move default build images to macos 15 (and xcode 16)

agocke opened the pull request:
This is to test our fixes for building using newer xcode and macos versions.

agocke committed:
Move default build images to macos 15 (and xcode 16)

This is to test our fixes for building using newer xcode and macos versions.

agocke committed:
Clarify build platform version requirements in YAML

Added comments to clarify build platform requirements.

agocke committed:
Merge branch 'main' into macos-15

agocke committed:
Merge branch 'main' into macos-15

akoeplinger commented:
yeah, we basically get away for end users because they tend to be on latest Xcode because Apple forces them.

I think we're in a bit of a pickle here. It looks like swift componetized a lot more of their implementation into libraries that get automatically linked in later versions, e.g. if you look at `libswift*` in `/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS.sdk/usr/lib/swift`

when you compile+link on the same machine this works fine but if you link on an older Xcode it depends on whether it shipped with all the libraries.

Given we're hitting the macos-13 brownout today we should probably disable the ios-arm64/tvos-arm64 jobs for now and merge this PR to unblock runtime.

The VMR already built with macos-15 / Xcode 16 in 10.0.1 so we already have that dependency there.

ericstj commented:
> We have XCode 14.3 on the helix machines which is old. We're trying to upgrade to the latest on the helix machines but we might not make it before the azdo deprecation date (4.12.).

> Given we're hitting the macos-13 brownout today we should probably disable the ios-arm64/tvos-arm64 jobs for now and merge this PR to unblock runtime.

Agreed - let's disable the tests since we can't get them fixed before the deprecation deadline.  That lets us hopefully get the build working before it all fails next week with official deprecation.  Who knows how to disable those tests?

ericstj committed:
Don't build on helix for mono+ios/tvos

Attempting to workaround xcode mismatch

ericstj committed:
Disable mono iOS/tvOS testing

Reverted "Don't build on helix for mono+ios/tvos" as we still were seeing failures

Comment out the test step entirely for now.

ericstj reviewed (approved):
Approving, but we should get review from others since I modified the pipeline.

@vitek-karas and @kotlarmilos are owners of iOS/tvOS

matouskozak committed:
disable swift-interop tests

kotlarmilos reviewed (approved):
LGTM, thanks!
